src/components/MultiLink/Legend.vue

```diff
@@ -123,105 +126,98 @@ export default Vue.extend({
           // eslint-disable-next-line @typescript-eslint/no-explicit-any
           let currentData: any[];
 
-          if (type === 'node') {
-            currentData = this.graphStructure.nodes.map((d: Node | Link) => d[attr]);
+          let xScale: ScaleLinear<number, number> | ScaleBand<string> | null = null;
```

Comment:
In general, I think doing it like this defeats the point of using TypeScript. In the rest of the code, you are using `as` to force the type of `xScale` to what it needs to be in each branch. Ideally, you would just declare `xScale` with the correct type in each context in which it's used.

I'm not saying you have to stop and fix this now, but you should in more general terms slow down your use of `as` to dispatch between types the way you're doing here. Essentially, `as` should be used as an escape hatch when there are no other ways to proceed, and you have some out-of-band reasoning for why it won't cause a problem. (In this example, you *do* have such reasoning, but there may still be other ways to express the code.)

Reply:
Agreed, I refactored this to use block scope vars inside the if and then I don't need to do all the forcing of types with `as`. I was trying to remove duplication of the axis appending code, but it makes sense for it to be duplicated -- they're affecting 2 different types of scales, numeric and categorical.

See 08c050c